src/Nri/Ui/Tabs/V5.elm

```diff
@@ -0,0 +1,341 @@
+module Nri.Ui.Tabs.V5 exposing
+    ( view
+    , Alignment(..)
+    , Tab, viewTabDefault
+    )
+
+{-|
+
+@docs view
+@docs Alignment
+@docs Tab, viewTabDefault
+
+-}
+
+import Accessibility.Styled.Aria as Aria
+import Accessibility.Styled.Role as Role
+import Accessibility.Styled.Widget as Widget
+import Css exposing (..)
+import EventExtras
+import Html.Styled as Html exposing (Attribute, Html)
+import Html.Styled.Attributes as Attributes
+import Html.Styled.Events as Events
+import Json.Decode
+import List.Zipper exposing (Zipper)
+import List.Zipper.Extra
+import Nri.Ui
+import Nri.Ui.Colors.Extra
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.Fonts.V1
+import Nri.Ui.Html.Attributes.V2 as AttributesExtra
+
+
+{-| Determines whether tabs are centered or floating to the left or right.
+-}
+type Alignment
+    = Left
+    | Center
+    | Right
+
+
+{-| -}
+type alias Tab id msg =
+    { id : id
+    , idString : String
```

Comment:
wouldn't it be neater to ask for a `id -> String` function in `view`?

Reply:
I think if we do the id -> String pattern, we would also want to do id -> Html msg for the tab and id -> Html msg for the tabpanel. My concern with that strategy is that you end up with the tab view separated from the panel view in the code, even though conceptually they're tied together.

Like I'd expect using the API to look like this:
```
type TabId
    = Passage
    | Feedback
    | TipsAndExamples
    | Rubric


idToString : TabId -> String
idToString tabId =
    case tabId of
        Passage ->
            "Passage"

        Feedback ->
            "Feedback"

        TipsAndExamples ->
            "TipsAndExamples"

        Rubric ->
            "Rubric"


viewTab : TabId -> Html Msg
viewTab tabId =
    case tabId of
        Passage ->
            text "Passage"

        Feedback ->
            text "Feedback"

        TipsAndExamples ->
            text "Tips and Examples"

        Rubric ->
            text "Rubric"


viewPanel : TabId -> Html Msg
viewPanel tabId =
    case tabId of
        Passage ->
            div []
                [ text "Passage"
                , text "blah blah"
                ]

        Feedback ->
            div [] [ "Feedback" ]

        TipsAndExamples ->
            div [] []

        Rubric ->
            div [] []

```

Versus:

```

allTabs =
    [ { id = Passage
      , idString = "passage"
      , spaHref = Nothing
      , tabView = Html.text "Passage"
      , panelView =
            div []
                [ text "Passage"
                , text "blah blah"
                ]
      }
    , { id = Feedback
      , idString = ""
      , spaHref = Nothing
      , tabView = Html.text "Feedback"
      , panelView = div [] [ "Feedback" ]
      }
    , { id = TipsAndExamples
      , idString = ""
      , spaHref = Nothing
      , tabView = Html.text "Tips & Examples"
      , panelView = div [] []
      }
    , { id = Rubric
      , idString = "rubric"
      , spaHref = Nothing
      , tabView = Html.text "Rubric"
      , panelView = div [] []
      }
    ]
```